src/nlp.jl

```diff
@@ -32,20 +59,54 @@ function getDual(c::ConstraintRef{NonlinearConstraint})
     return nldata.nlconstrDuals[c.idx]
 end
 
+type FunctionStorage
+    nd::Vector{NodeData}
+    adj::SparseMatrixCSC{Bool,Int}
+    const_values::Vector{Float64}
+    forward_storage::Vector{Float64}
+    reverse_storage::Vector{Float64}
+    grad_sparsity::Vector{Int}
+    hess_I::Vector{Int} # nonzero pattern of hessian
+    hess_J::Vector{Int}
+    rinfo::Coloring.RecoveryInfo # coloring info for hessians
+    seed_matrix::Matrix{Float64}
+    linearity::Linearity
+    dependent_subexpressions::Vector{Int} # subexpressions which this function depends on, ordered for forward pass
+end
+
+type SubexpressionStorage
+    nd::Vector{NodeData}
+    adj::SparseMatrixCSC{Bool,Int}
+    const_values::Vector{Float64}
+    forward_storage::Vector{Float64}
+    reverse_storage::Vector{Float64}
+    forward_hessian_storage::Vector{Dual{Float64}}
+    reverse_hessian_storage::Vector{Dual{Float64}}
+end
+
 type JuMPNLPEvaluator <: MathProgBase.AbstractNLPEvaluator
     m::Model
     A::SparseMatrixCSC{Float64,Int} # linear constraint matrix
+    parameter_values::Vector{Float64}
     has_nlobj::Bool
     linobj::Vector{Float64}
-    eval_f_nl::Function
-    eval_fgrad_nl::Function
-    eval_h_nl::Function
-    nnz_hess_obj::Int
-    jac_I::Vector{Int}
-    jac_J::Vector{Int}
-    hess_I::Vector{Int}
-    hess_J::Vector{Int}
-    requested_hessian::Bool
+    objective::FunctionStorage
+    constraints::Vector{FunctionStorage}
+    subexpressions::Vector{SubexpressionStorage}
+    subexpression_order::Vector{Int}
+    subexpression_forward_values::Vector{Float64}
+    subexpression_reverse_values::Vector{Float64}
+    subexpressions_as_julia_expressions::Vector{Any}
+    last_x::Vector{Float64}
+    jac_storage::Vector{Float64} # temporary storage for computing jacobians
+    # storage for computing hessians
```

Comment:
This is a comment about the following 7 or so lines, not the `want_hess`, if thats what you mean